File: 백준/Silver/16953. A → B/A → B.cs
using System;
class Program
{
    static void Main()
    {
        int[] numberArray = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();
        int number1 = numberArray[0];
        int number2 = numberArray[1];
        int count = 0;
        
        while(number1!=number2 && number2 >0)
        {
            if(number2 % 2 == 0)
            {
                number2 /= 2;
            }
            else if(number2 % 10 == 1)
            {
                number2 = number2 / 10;
            }
            else
            {
                break;
            }
            count++;
        }
        if(number1 != number2)
        {
            Console.WriteLine(-1);
        }
        else
        {
            Console.WriteLine(count+1);
        }
    }
}

File: 백준/Silver/16953. A → B/A → B.kt
import java.io.*

fun main(){
    val br = BufferedReader(InputStreamReader(System.`in`))
    val (a,b) = br.readLine().split(' ').map{ it.toLong() }
    var current = b
    var count = 0
    while(current > a){
        if(current % 2L == 0L){
            current /= 2
        } else if (current % 10L == 1L){
            current /= 10
        } else {
            break
        }
        count++
    }
    
    if (current == a){
        println(count + 1)
    } else {
        println(-1)
    }
}